astropy/io/misc/hdf5.py

```diff
@@ -282,10 +302,12 @@ def write_table_hdf5(table, output, path=None, compression=False,
 
         # Open the file for appending or writing
         f = h5py.File(output, 'a' if append else 'w')
-
+        for column in table.colnames:
+            if hasattr(table.columns[column],"dtype"):
+                if table.columns[column].dtype.kind=='U': serialize_meta=True
```

Comment:
Why adding this ? Maybe for the `__is_encoded_unicode__` meta, in which case you can remove this.

Reply:
Yes without this the required meta data is not stored.